main.py

```diff
@@ -125,13 +125,18 @@ async def on_typing(channel, user, when):
     logging.debug('Typing from {0.name}'.format(user))
 
 @mbox.event
-async def on_guild_join(guild):
+async def on_guild_join(guild: discord.Guild):
     logging.info('Joined Server: {0.name}'.format(guild))
-    await guild.text_channels[0].send('Thanks for adding Music Bot!')
-    await src.preinitialization.generate_profile(guild, mbox, profiles)
-    for profile in profiles:
-        await profile.setup()
-    print(len(profiles))
+    try:
+        await guild.text_channels[0].send('Thanks for adding Music Bot!')
+        await src.preinitialization.generate_profile(guild, mbox, profiles)
+        for profile in profiles:
+            await profile.setup()
+    except discord.errors.Forbidden:
```

Comment:
Are 403's the only error we see due to not having permissions? How about 401's?

Reply:
Yea I think 401 is also something we definitely want to have

Reply:
It looks like 403's are the only one that it gives when there is a permission error.
https://discord.com/developers/docs/topics/opcodes-and-status-codes#http
This is why I think the discord python library gives this 403 its own custom error instead of a generic discord.HTTPException
https://discordpy.readthedocs.io/en/stable/api.html?highlight=forbidden#discord.Forbidden

Reply:
ok got it. Seems like it doesn't define logic related to 401's then. Thanks for clarifying